### PR TITLE
[IT520] - CardDav synchronization - using the getDBValue method

### PR DESCRIPTION
**Pull request by arek1-618**
[IT520] - CardDav synchronization - using the getDBValue method

### CODE REVIEW

File: modules/API/models/CardDAV.php [hunk 1]
```diff
@@ -277,48 +277,64 @@ public function deletedCard($card)
...
-		$leadId = false;
+		$record = Vtiger_Record_Model::getCleanInstance($moduleName);
 		if (isset($vcard->ORG)) {
 			$lead = Vtiger_Record_Model::getCleanInstance('Leads');
 			$lead->set('assigned_user_id', $this->user->get('id'));
-			$lead->set('company', \App\Purifier::purify((string) $vcard->ORG));
-			$lead->set('lastname', \App\Purifier::purify((string) $vcard->ORG));
+			$orgName = $lead->getField('company')->getDBValue(\App\Purifier::purify((string) $vcard->ORG));
+			$lead->set('company', $orgName);
+			$lead->set('firstname', $orgName);
```
Comment (rskrzypczak) [inline modules/API/models/CardDAV.php]: ?

File: modules/Vtiger/models/Record.php [hunk 2]
```diff
@@ -90,6 +90,19 @@ public function setFromUserValue($fieldName, $value)
...
 
+	/**
+	 * Set the value for the database.
+	 *
+	 * @param string $key
+	 * @param mixed $value
+	 *
+	 * @return void
+	 */
+	public function setDBValue(string $key, $value)
```
Comment (rskrzypczak) [inline modules/Vtiger/models/Record.php]: remove

File: modules/API/models/CardDAV.php [hunk 3]
```diff
@@ -589,17 +591,17 @@ public function setRecordAddres(Sabre\VObject\Component $vcard, $moduleName, Vti
 				$adr = $property->getParts();
 				$street = $adr[1] . ' ' . $adr[2];
 				if ('Contacts' === $moduleName) {
-					$record->set('addresslevel1' . $contactsPostFix, \App\Fields\Country::findCountryName(\App\Purifier::purify($adr[6]))); //country
-					$record->set('addresslevel7' . $contactsPostFix, \App\Purifier::purify($adr[5])); //code
-					$record->set('addresslevel2' . $contactsPostFix, \App\Purifier::purify($adr[4])); //state
-					$record->set('addresslevel5' . $contactsPostFix, \App\Purifier::purify($adr[3])); //city
-					$record->set('addresslevel8' . $contactsPostFix, \App\Purifier::purify(trim($street))); //street
+					$record->setDBValue('addresslevel1' . $contactsPostFix, \App\Fields\Country::findCountryName(\App\Purifier::purify($adr[6]))); //country
+					$record->setDBValue('addresslevel7' . $contactsPostFix, \App\Purifier::purify($adr[5])); //code
```
Comment (rskrzypczak) [inline modules/API/models/CardDAV.php]: remove all comments

File: modules/API/models/CardDAV.php [hunk 4]
```diff
@@ -12,12 +12,54 @@ class API_CardDAV_Model
 	const ADDRESSBOOK_NAME = 'YFAddressBook';
 	const PRODID = 'YetiForceCRM';
 
+	/**
+	 * Address mapping for modules.
+	 */
+	const ADDRESS_MAPPING = [
```
Comment (mariuszkrzaczkowski) [inline modules/API/models/CardDAV.php]: move to App\Integrations\Dav\Card

File: modules/API/models/CardDAV.php [hunk 5]
```diff
@@ -277,48 +319,64 @@ public function deletedCard($card)
...
+	 * @return void
+	 */
+	public function createRecord(string $moduleName, array $card)
 	{
 		\App\Log::trace(__METHOD__ . ' | Start Card ID' . $card['id']);
 		$vcard = Sabre\VObject\Reader::read($card['carddata']);
-		$leadId = false;
-		if (isset($vcard->ORG)) {
+		$record = Vtiger_Record_Model::getCleanInstance($moduleName);
+		if ($moduleName==='Contacts' && isset($vcard->ORG)) {
```
Comment (mariuszkrzaczkowski) [inline modules/API/models/CardDAV.php]: PHPCS

File: modules/API/models/CardDAV.php [hunk 6]
```diff
@@ -277,48 +319,64 @@ public function deletedCard($card)
...
+	/**
+	 * Update record.
+	 *
+	 * @param Vtiger_Record_Model $record
+	 * @param array               $card
+	 */
+	public function updateRecord(Vtiger_Record_Model $record, array $card)
+	{
+		\App\Log::trace(__METHOD__ . ' | Start Card ID:' . $card['id']);
+		$vcard = Sabre\VObject\Reader::read($card['carddata']);
```
Comment (mariuszkrzaczkowski) [inline modules/API/models/CardDAV.php]: move to App\Integrations\Dav\Card::loadFromContent
similar to App\Integrations\Dav\Calendar::loadFromContent

File: modules/Vtiger/models/Record.php [hunk 7]
```diff
@@ -90,6 +90,22 @@ public function setFromUserValue($fieldName, $value)
...
```
Comment (mariuszkrzaczkowski) [inline modules/Vtiger/models/Record.php]: setDBValue > setFromDisplayFormat

File: modules/Vtiger/models/Record.php [hunk 8]
```diff
@@ -90,6 +90,22 @@ public function setFromUserValue($fieldName, $value)
...
+	{
+		$fieldModel = $this->getField($key);
+		if($fieldModel){
```
Comment (mariuszkrzaczkowski) [inline modules/Vtiger/models/Record.php]:
```suggestion
		if($fieldModel = $this->getField($key)){
```

**Pull request closed without merging**